tools/cldr-code/src/main/java/org/unicode/cldr/util/MatchValue.java

```diff
@@ -317,28 +418,39 @@ private ValidityMatchValue(LstrType type, Set<Status> statuses, boolean shortId)
                 throw new IllegalArgumentException("short- not supported except for units");
             }
             this.shortId = shortId;
+            // validForType = Validity.getInstance().getStatusToCodes(type).keySet();
             this.statuses =
-                    statuses == null ? EnumSet.allOf(Status.class) : ImmutableSet.copyOf(statuses);
+                    ImmutableSet.copyOf(statuses == null ? DEFAULT_STATUS.get(type) : statuses);
         }
 
         public static MatchValue of(String typeName) {
             if (typeName.equals("locale")) {
                 return new LocaleMatchValue();
             }
+            if (typeName.equals("xlocale")) {
+                return new XLocaleMatchValue();
```

Comment:
Instead of confusing variable names `xlocale` and `nlocale` can we spell it out? And since we can use logical ors we can just make specific sets eg. `deprecatedlocale` and `overlonglocale` and people use like`validity/locale||validity/overlonglocale`.

Reply:
What I intended was having common semantics nlocale doesn't mean 'or deprecated'), but nlocale and xlocale are obscure. I'll change to more meaningful names.